### PR TITLE
Sprint/s159

### PR DESCRIPTION
**Pull request by ffaraoneim**
### Major changes (unordered)
* New class APIError that allow developers to quickly access the JSON error object returned by Connect.
* Upgrade to python 3.7 and sphinx 2.3.0 to build rtd docs.
* Get version for documentation from package.json
* add support for filtering, paging and ordering on the requests list method
* change the ConversationService class to remove "business logic"
* new class Fulfillment, a sort of façade on top of api access: some methods have been moved from the xxxxService classes to the Fulfillment class.
* rename of some method in the xxxxService classes

### CODE REVIEW

File: lib/connect/api/conversations.js [hunk 1]
```diff
@@ -12,20 +12,20 @@ const BaseService = require('./base');
...
-   * Retrieves the id of the conversation of request.
+   * Retrieves the conversation of the conversation of instance.
    *
    * @param {string} id   The unique request identifier for which retrieve the conversationId.
    *
    * @returns {string}    Conversation ID.
    */
 
-  async getConversationIdByRequest(id) {
+  async getConversation(id) {
```
Comment (ffaraoneim) [inline lib/connect/api/conversations.js]: The getConversation method returns a conversation based on the "instance_id" parameter where instance_id is the ID of the object attached to the conversation. I think a better name could be getConversationByObjectId or similar.

File: lib/connect/api/conversations.js [hunk 2]
```diff
@@ -12,20 +12,20 @@ const BaseService = require('./base');
...
     const url = this.addParams('/conversations', { instance_id: id });
     const response = await this.fetch(url);
     this.checkResponse(response);
     const conv = await response.json();
     if (conv.length === 1) {
```
Comment (ffaraoneim) [inline lib/connect/api/conversations.js]: this if is unnecessary: return response.json() instead.

File: lib/connect/api/conversations.js [hunk 3]
```diff
@@ -53,24 +52,21 @@ class ConversationService extends BaseService {
...
    *
    * @param {string} id   The unique request identifier for which publish.
    *
    * @returns {object} An object with the messages.
    */
   async getMessages(id) {
-    const conversationId = await this.getConversationIdByRequest(id);
-    const url = `/conversations/${conversationId}`;
+    const url = `/conversations/${id}`;
     const options = {
```
Comment (ffaraoneim) [inline lib/connect/api/conversations.js]: The "GET" http method is the default method. Please remove the constant "options" and invoke fetch with only the url argument.

File: lib/connect/api/conversations.js (hunk 1 above)
Comment (marcserrat) [reply lib/connect/api/conversations.js]: agree

File: lib/connect/api/errors.js [hunk 4]
```diff
@@ -16,8 +16,48 @@ class HttpError extends Error {
...
+ *
+ *
+ */
+  constructor(status, message) {
+    super(status, message);
+    /**
+     * The JSON error object.
+     * @type {Object}
+     */
+    this.json = JSON.parse(message);
```
Comment (AntonHinz) [inline lib/connect/api/errors.js]: [MINOR] For me `json` is too abstract maybe we can follow ES here and name it `toJSON`

**Pull request merged**